core/src/main/java/io/confluent/rest/SslFactory.java

```diff
@@ -103,27 +106,22 @@ private static FileWatcher.Callback onFileChangeCallback(SslConfig sslConfig,
     };
   }
 
+
   public static SslContextFactory createSslContextFactory(SslConfig sslConfig) {
+    return createSslContextFactory(sslConfig, null);
+  }
+
+  public static SslContextFactory createSslContextFactory(
+      SslConfig sslConfig,
+      X509Source x509Source) {
     SslContextFactory.Server sslContextFactory = new SslContextFactory.Server();
+    
+    if (sslConfig.getIsSpireEnabled()) {
+      configureSpiffeSslContext(sslContextFactory, x509Source);
+    }
 
     if (!sslConfig.getKeyStorePath().isEmpty()) {
-      setSecurityStoreProps(sslConfig, sslContextFactory, true, false);
-      sslContextFactory.setKeyManagerPassword(sslConfig.getKeyManagerPassword());
-
-      if (!sslConfig.getKeyManagerFactoryAlgorithm().isEmpty()) {
-        sslContextFactory.setKeyManagerFactoryAlgorithm(sslConfig.getKeyManagerFactoryAlgorithm());
-      }
-
-      if (sslConfig.getReloadOnKeyStoreChange()) {
-        Path watchLocation = Paths.get(sslConfig.getReloadOnKeyStoreChangePath());
-        try {
-          FileWatcher.onFileChange(watchLocation,
-              onFileChangeCallback(sslConfig, sslContextFactory));
-          log.info("Enabled SSL cert auto reload for: " + watchLocation);
-        } catch (java.io.IOException e) {
-          log.error("Cannot enable SSL cert auto reload", e);
-        }
-      }
+      configureKeyStore(sslContextFactory, sslConfig);
```

Comment:
Both shouldn't be configured right?

Reply:
if we have `spire` enabled,  you are right that they won't be needed to be configured; However, even configuring won't do much anything because internally the server will go to talk spire agent to grab all the trust bundle and stuff into cache; it will never have a need to reach the KeyStore even Keystore was configured. I already provided the comment [here](https://github.com/confluentinc/rest-utils/blob/6e48e15ceb60867f03773d610e1f4c3c684564e8/core/src/main/java/io/confluent/rest/RestConfig.java#L211)